src/Microsoft.AspNet.Http/Authentication/DefaultAuthenticationManager.cs

```diff
@@ -0,0 +1,147 @@
+
+namespace Microsoft.AspNet.Http.Authentication
+{
+    public class DefaultAuthenticationManager : AuthenticationManager
+    {
+        private readonly IFeatureCollection _features;
+        private FeatureReference<IHttpAuthenticationFeature> _authentication = FeatureReference<IHttpAuthenticationFeature>.Default;
+        private FeatureReference<IHttpResponseFeature> _response = FeatureReference<IHttpResponseFeature>.Default;
+
+        public DefaultAuthenticationManager(IFeatureCollection features)
+        {
+            _features = features;
+        }
+
+        private IHttpAuthenticationFeature HttpAuthenticationFeature
+        {
+            get { return _authentication.Fetch(_features) ?? _authentication.Update(_features, new HttpAuthenticationFeature()); }
+        }
+
+        private IHttpResponseFeature HttpResponseFeature
+        {
+            get { return _response.Fetch(_features); }
+        }
+
+        public override IEnumerable<AuthenticationDescription> GetAuthenticationSchemes()
+        {
+            var handler = HttpAuthenticationFeature.Handler;
+            if (handler == null)
+            {
+                return new AuthenticationDescription[0];
+            }
+
+            var describeContext = new DescribeSchemesContext();
+            handler.GetDescriptions(describeContext);
+            return describeContext.Results.Select(description => new AuthenticationDescription(description));
+        }
+
+        public override AuthenticationResult Authenticate([NotNull] string authenticationScheme)
+        {
+            var handler = HttpAuthenticationFeature.Handler;
+
+            var authenticateContext = new AuthenticateContext(authenticationScheme);
+            if (handler != null)
+            {
+                handler.Authenticate(authenticateContext);
+            }
+
+            if (!authenticateContext.Accepted)
+            {
+                throw new InvalidOperationException("The following authentication scheme was not accepted: " + authenticationScheme);
+            }
+
+            return new AuthenticationResult(authenticateContext.Principal,
+                new AuthenticationProperties(authenticateContext.Properties),
+                new AuthenticationDescription(authenticateContext.Description));
+        }
+
+        public override async Task<AuthenticationResult> AuthenticateAsync([NotNull] string authenticationScheme)
+        {
+            var handler = HttpAuthenticationFeature.Handler;
+
+            var authenticateContext = new AuthenticateContext(authenticationScheme);
+            if (handler != null)
+            {
+                await handler.AuthenticateAsync(authenticateContext);
+            }
+
+            // Verify all types ack'd
+            if (!authenticateContext.Accepted)
+            {
+                throw new InvalidOperationException("The following authentication scheme was not accepted: " + authenticationScheme);
```

Comment:
Nit: $"The following.... {authenticationScheme}.."

Ditto other exception messages below.